build/package/Installer.RPM.targets

```diff
@@ -242,12 +237,16 @@
                 ToolPath="$(RpmInstalledDirectory)" />
 
     <!-- Clean up Packages -->
-    <Exec Command="sudo yum remove -y $(SdkRpmPackageName)" />
+    <ItemGroup>
+      <InstalledPackageName Include="$(SdkInstallerFile)" />
+      <InstalledPackageName Include="$(DownloadedAspNetCoreSharedFxInstallerFile)" />
+      <InstalledPackageName Include="$(DownloadedSharedFrameworkInstallerFile)" />
+      <InstalledPackageName Include="$(DownloadedHostFxrInstallerFile)" />
+      <InstalledPackageName Include="$(DownloadedSharedHostInstallerFile)" />
+      <InstalledPackageName Include="$(DownloadedRuntimeDepsInstallerFile)" />
+    </ItemGroup>
 
-    <Exec Command="sudo yum remove -y $(AspNetCoreSharedFxRpmPackageFileName)" />
-    <Exec Command="sudo yum remove -y $(SharedFxRpmPackageFileName)" />
-    <Exec Command="sudo yum remove -y $(HostFxrRpmPackageName)" />
-    <Exec Command="sudo yum remove -y $(HostRpmPackageName)" />
-    <Exec Command="sudo yum remove -y $(RuntimeDepsPackageFileName)" />
+    <!-- Query the package name of the file and give it to uninstall input -->
+    <Exec Command="sudo rpm -e `rpm --queryformat &quot;%25{NAME}&quot; -qp %(InstalledPackageName.Identity)`" />
```

Comment:
More error pop up due to rpm cannot find the correct package name to remove. I replaced "guessing" package name code to query package name from the file.

Reply:
rpm package name dotnet-runtime-deps-2.1
rpm file name dotnet-runtime-deps-2.1.3-rhel.7-x64.rpm  

we have logic to construct rpm package name from rpm file name